frame/contracts/src/migration.rs

```diff
@@ -15,7 +15,46 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Migration framework for pallets.
+//! Multi-block Migration framework for pallet-contracts.
+//!
+//! This module allows us to define a migration as a sequence of [`MigrationStep`]s that can be
+//! executed across multiple blocks.
+//!
+//! # Usage
+//!
+//! A migration step is defined under `src/migration/vX.rs`, where `X` is the version number.
+//! For example, `vX.rs` defines a migration from version X to version X + 1.
+//!
+//! ## Example:
+//!
+//! To configure a migration to v12 for a runtime using v9 of `pallet-contracts` on the chain, you
+//! would set the `Migrations` trait as follows:
+//!
+//! ```
+//! use pallet_contracts::migration::{v9, v10, v11};
+//! # pub enum Runtime {};
+//! type Migrations = (v9::Migration<Runtime>, v10::Migration<Runtime>, v11::Migration<Runtime>);
+//! ```
+//!
+//! ## Notes:
+//!
+//! - Migrations should always be tested with `try-runtime` before being deployed.
+//! - By testing with `try-runtime` against a live network, you ensure that all migration steps work
+//!   and that you have included the required steps.
+//!
+//! ## Low Level / Implementation Details
+//!
+//! When a migration starts and [`OnRuntimeUpgrade::on_runtime_upgrade`] is called, instead of
```

Comment:
link doesn't work

Reply:
mmm It works for me 🤔 
![Screenshot 2023-06-08 at 15 10 36](https://github.com/paritytech/substrate/assets/521091/299b376d-cad0-4f71-81d7-fd738e757cc1)

Reply:
yeah sorry, it was just built  with `---no-deps`

> MigrationInProgress is private so you can't link to it

why not, you can if build it with `--document-private-items`. btw there are a couple of broken links in that case:

```
warning: unresolved link to `ReservableCurrency`
  --> frame/contracts/src/storage/meter.rs:93:15
   |
93 | /// It uses [`ReservableCurrency`] in order to do accomplish the reserves.
   |               ^^^^^^^^^^^^^^^^^^ no item named `ReservableCurrency` in scope
   |
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
   = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default

warning: unresolved link to `T::Migrations`
   --> frame/contracts/src/migration.rs:337:49
    |
337 |     /// Verify that each migration's step of the [`T::Migrations`] sequence fits into `Cursor`.
    |                                                    ^^^^^^^^^^^^^ no item named `T` in scope

warning: `pallet-contracts` (lib doc) generated 2 warnings
```

Reply:
mm is that useful at all, since we don't include that in the doc hosted [here](https://paritytech.github.io/substrate/master/pallet_contracts/struct.Migration.html)

Or are you saying that fixing that would help navigate code from your IDE?

Reply:
> mm is that useful at all, since we don't include that in the doc hosted [here](https://paritytech.github.io/substrate/master/pallet_contracts/struct.Migration.html)

For sure it would be useful for developers working on it. At the end of the day, once we document private items, we need to have their docs consistent/properly linked.

Reply:
right it does highlights stuff properly in the code, fixed the warnings

Reply:
Well, this is quite unhandy! Because this warning does not make sense to me. It arises both when building with `--document-private-items` flag and without it. Still, the link won't get generated if building w\o this flag, which is what we want, right? So probably it worth adding `#![allow(rustdoc::private_intra_doc_links)]` to `src/lib.rs`